### PR TITLE
Refactor and update databricks integration page

### PR DESCRIPTION
**Pull request by liangz1**
# Your PR Title (What it does)

Fixes # (issue)

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

### CODE REVIEW

**Comment by dev2049** [comment]
@liangz1 lgtm, just lemme know when it's ready

File: docs/integrations/databricks.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+Databricks
+==========
+
+The [Databricks](https://www.databricks.com/) Lakehouse Platform unifies data, analytics, and AI on one platform.
+
+Databricks embraces the LangChain ecosystem in various ways:
+
+1. Databricks connector for the SQLDatabase Chain: SQLDatabase.from_databricks() provides an easy way to query your data on Databricks through LangChain
+2. Databricks-managed MLflow integrates with LangChain: Tracking and serving LangChain applications with fewer steps
+3. Databricks as an LLM provider: Deploy your fine-tuned LLMs on Databricks and query the endpoint as langchain.llms.Databricks
```
Comment (epec254) [inline docs/integrations/databricks.md:10]: This also supports Driver Proxy for development - lets add this reference, pls

File: docs/integrations/databricks.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Databricks-managed MLflow integrates with LangChain
+---------------------------------------------------
+
+MLflow is an open source platform to manage the ML lifecycle, including experimentation, reproducibility, deployment, and a central model registry. See the notebook [MLflow Callback Handler](./mlflow_tracking.ipynb) for details about MLflow's integration with LangChain.
+
+Databricks provides a fully managed and hosted version of MLflow integrated with enterprise security features, high availability, and other Databricks workspace features such as experiment and run management and notebook revision capture. MLflow on Databricks offers an integrated experience for tracking and securing machine learning model training runs and running machine learning projects. See [MLflow guide](https://docs.databricks.com/mlflow/index.html) for more details.
+
+Databricks-managed MLflow makes it more convenient to develop LangChain applications on Databricks. For MLflow tracking, you don't need to set the tracking uri. For MLflow Model Serving, you can save LangChain Chains in the MLflow langchain flavor, and then register and serve the Chain with a few clicks on Databricks.
```
Comment (epec254) [inline docs/integrations/databricks.md:24]: Let's also reference the ability of Model Serving to securely manage credentials.

File: docs/integrations/databricks.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Databricks as an LLM provider
+-----------------------------
+
+The notebook [Wrap Databricks endpoints as LLMs](../modules/models/llms/integrations/databricks.html) illustrates the method to wrap Databricks endpoints as LLMs in LangChain. It supports two types of endpoints: the serving endpoint, which is recommended for both production and development, and the cluster driver proxy app, which is recommended for interactive development. 
+
+Databricks Dolly
```
Comment (epec254) [inline docs/integrations/databricks.md:31]: I would clarify that Databricks endpoints support Dolly, but are also great for hosting models like MPT-7B or any others from the HF ecosystem.  They also support hosting proprietary models like OpenAI to provide a  governance layer for enterprises.

File: docs/integrations/databricks.md (hunk 1 above)
Comment (liangz1) [reply docs/integrations/databricks.md]: added

File: docs/integrations/databricks.md (hunk 2 above)
Comment (liangz1) [reply docs/integrations/databricks.md]: added

**Comment by liangz1** [comment]
@epec254 Thanks for the feedback. I added the suggested points. Let me know if anything you would like to add.

**Comment by liangz1** [comment]
Hi @dev2049 , this is ready to merge. Thanks!

**Pull request merged**